urllib3/util/retry.py

```diff
@@ -330,7 +338,7 @@ def increment(self, method=None, url=None, response=None, error=None,
 
         elif error and self._is_read_error(error):
             # Read retry?
-            if read is False:
+            if read is False or (method is not None and not self._is_method_retryable(method)):
```

Comment:
@Lukasa That seems ok to me! I'm thinking of just adding `method='GET'` (or equivalent) to the tests that need it, is that what you have in mind?